### DISCUSSION TITLE
Suggestion

### DISCUSSION THREAD

**Discussion by Vishwas-verma** [adonisjs/core#2893] [Help]
```
  async index(): Promise<User[]> {
    return User.query().select("id","follow_up","name").groupBy(['follow_up']);
  }
```

Hye this simple query is not grouping by follow_up? The issue I am getting is 
```
select `id`, `follow_up`, `name` from `users` group by `follow_up` - ER_WRONG_FIELD_WITH_GROUP: Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'chromeDB.users.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by

```

Any Suggestion?

**Comment by varun-nambiar** [comment]
https://www.sitepoint.com/quick-tip-how-to-permanently-change-sql-mode-in-mysql/

I usually add just the following line in mysqld.cnf
(and restart mysql service)

```
sql-mode=

```

**Reply by Vishwas-verma** [reply]
Hey do you have any other way as I dont think its not the right thing to do.

**Reply by varun-nambiar** [reply]
You have to remove 'ONLY_FULL_GROUP_BY' from the sql_mode variable
on Run time after connecting to mysql you will have to run the following 
```sql
  SET SESSION sql_mode = CONCAT(REPLACE(@@sql_mode, ',ONLY_FULL_GROUP_BY', ''));
```

In Laravel/Codeigniter we had an option to set this on the connection settings in the framework config\
There should be something similar for adonis too in the config\
Or else you can run the above query once on boot (AppProvider?) to remove `ONLY_FULL_GROUP_BY`

**Comment by McSneaky** [comment]
> ```
>   async index(): Promise<User[]> {
>     return User.query().select("id","follow_up","name").groupBy(['follow_up']);
>   }
> ```
> 
> Hye this simple query is not grouping by follow_up? The issue I am getting is
> 
> ```
> select `id`, `follow_up`, `name` from `users` group by `follow_up` - ER_WRONG_FIELD_WITH_GROUP: Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'chromeDB.users.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
> ```
> 
> Any Suggestion?

Question is. What to do with user IDs? If it groups them by follow up, but it has multiple user IDs, then it restults nested array. Something like this:
node ace configure @adonisjs/auth

|user.id | user.follow_up | user.name|
|---|---|---|
|[1, 40, 2] | 1 | "McSneaky"|
| 4 | 0 | "Someone Else"|

It cant group normally, coz there might be multiple user IDs per single follow_up

You can add ID to groupBy()

```js
return User.query().select("id","follow_up","name").groupBy(['follow_up', 'id']);
```

**Reply by Vishwas-verma** [reply]
What I want to achieve is array of names,ids ,corresponding to follow_up date ?

**Reply by Vishwas-verma** [reply]
> ```js
> ```js
> return User.query().select("id","follow_up","name").groupBy(['follow_up', 'id']);
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
>         
> ```

Hello @McSneaky ,
What I am recieving by this query 
```
return User.query().select("id","follow_up","name").groupBy(['follow_up', 'id']);
```
is-
```
{
    "data": [
        {
            "id": 1,
            "follow_up": "2021-07-09",
            "name": "Varun"
        },
        {
            "id": 2,
            "follow_up": "2021-07-29",
            "name": "Vishs"
        },
        {
            "id": 3,
            "follow_up": "2021-07-10",
            "name": "Fs"
        },
        {
            "id": 4,
            "follow_up": "2021-07-09",
            "name": "Appukd"
        },
        {
            "id": 5,
            "follow_up": "2021-07-10",
            "name": "fdf"
        }
    ]
}
```

I want to group so that all users with same follow_up date come together.

**Reply by McSneaky** [reply]
That's explaining visually what's going on: https://stackoverflow.com/a/38551525

If you disable `ONLY_FULL_GROUP_BY` then it's same as using `ANY_VALUE()` which isn't a thing you want

Most likely you should make multiple queries or use `GROUP_CONCAT()`. 
Here's good example how to use it: https://www.geeksforgeeks.org/mysql-group_concat-function/


Easiest to get users with same `follow_up` date is to do it like this:
```js
return User.query().select("id","follow_up","name").where({ follow_up: "2021-07-10" });
```

Depends on data size. Might want to first select all follow ups and then fire off query with all those dates. It generates quite some requests tho
